open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/validate/VRPSubmissionTest.java

```diff
@@ -0,0 +1,888 @@
+
+    @BeforeClass
+    public void initClass() throws ReflectiveOperationException {
+        MockitoAnnotations.initMocks(this);
+
+        //to execute util class initialization
+        new CarbonUtils();
+        System.setProperty("some.property", "property.value");
+        System.setProperty("carbon.home", ".");
+        ConsentExtensionTestUtils.injectEnvironmentVariable("CARBON_HOME", ".");
+
+        configMap = new HashMap<>();
+        configMap.put("ErrorURL", "https://localhost:8243/error");
+
+        consentValidator = new DefaultConsentValidator();
+        consentValidateDataMock = mock(ConsentValidateData.class);
+        authorizationResources.add(getAuthorizationResource());
+        detailedConsentResourceMock = mock(DetailedConsentResource.class);
+        consentCoreServiceMock = mock(ConsentCoreServiceImpl.class);
+    }
+
+    @BeforeMethod
+    public void initMethod() {
+
+        OpenBankingConfigParser openBankingConfigParserMock =  mock(OpenBankingConfigParser.class);
+        doReturn(configMap).when(openBankingConfigParserMock).getConfiguration();
+
+        PowerMockito.mockStatic(OpenBankingConfigParser.class);
+        PowerMockito.when(OpenBankingConfigParser.getInstance()).thenReturn(openBankingConfigParserMock);
+
+        OpenBankingConfigParser openBankingConfigParsersMock =  mock(OpenBankingConfigParser.class);
+        doReturn(configMap).when(openBankingConfigParserMock).getConfiguration();
+
+        PowerMockito.mockStatic(OpenBankingConfigParser.class);
+        PowerMockito.when(OpenBankingConfigParser.getInstance()).thenReturn(openBankingConfigParsersMock);
+    }
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+    @Test
+    public void testValidateInitiation() throws ParseException {
+
+        JSONObject initPayload = ConsentExtensionTestUtils.getJsonPayload(
+                ConsentValidateTestConstants.VRP_INITIATION);
+        JSONObject subPayload = ConsentExtensionTestUtils.getJsonPayload(
+                ConsentValidateTestConstants.VRP_SUBMISSION);
+
+        JSONObject validationResult = validator.validateInitiation(
+                ConsentExtensionTestUtils.getInitiationPayload(subPayload),
+                ConsentExtensionTestUtils.getInitiationPayload(initPayload));
+
+        Assert.assertTrue((Boolean) validationResult.get(ConsentExtensionConstants.IS_VALID_PAYLOAD));
+    }
+
+    @Test
+    public void testValidateInstruction() throws ParseException {
+
+        JSONObject initPayload = ConsentExtensionTestUtils.getJsonPayload(
+                ConsentValidateTestConstants.VRP_INITIATION_WITHOUT_CREDITOR_ACC);
+        JSONObject subPayload = ConsentExtensionTestUtils.getJsonPayload(
+                ConsentValidateTestConstants.VRP_SUBMISSION);
+
+        JSONObject validationResult = validator.validateCreditorAcc(
```

Comment:
Is testValidateInstruction is cover by `validateCreditorAcc`?

Reply:
Fixed in [bfaf99e]